src/char/reader.cr

```diff
@@ -27,10 +47,25 @@ struct Char
       decode_current_char
     end
 
+    # Returns true if there is a character left to read.
+    # The terminating byte '\0' is considered a valid character
+    # by this method.
+    # ```
+    # reader = Char::Reader.new("a")
+    # reader.has_next? #=> true
+    # reader.peek_next_char #=> '\0'
+    # ```
     def has_next?
       !@end
     end
 
+    # Reads the next character in the `#string`,
```

Comment:
I'm not sure to reference in the method description another instance method. Since is not an argument, you cannot use `*string*` (that is the standard).

Perhaps just say: *Reads the next character in the string* without any formatting for what *string* is?